src/Native/LdaNative/model_block.cpp

```diff
@@ -113,20 +124,22 @@ namespace lda
         dict_ = new WordEntry[num_vocabs_];
         for (int i = 0; i < num_vocabs_; ++i)
         {
-            // This warning is a false positive. Supressing it similar to the existing one on Line 140 below.
-#pragma warning(suppress: 6386)
             dict_[i].is_dense_ = 0;
             dict_[i].is_alias_dense_ = 0;
         }
 
-        mem_block_size_ = mem_block_size;
+		if (mem_block_size < 0 || static_cast<uint64_t>(mem_block_size) > numeric_limits<size_t>::max())
+			throw bad_alloc();
+        mem_block_size_ = static_cast<size_t>(mem_block_size);
         mem_block_ = new int32_t[mem_block_size_]();   // NOTE : force to initialize the values to be zero
 
-        alias_mem_block_size_ = alias_mem_block_size;
+		if (mem_block_size < 0 || static_cast<uint64_t>(alias_mem_block_size) > numeric_limits<size_t>::max())
```

Comment:
>alias_mem_block_size [](start = 50, length = 20)

I noticed when outputting the size, you are multiplying by 4. Should the *4 apply here in the check? And what is the *4? Is this supposed to be sizeof(float)?

Reply:
I think that is the size of the array item (sizeof(int32_t) I'll update

---
In reply to: [276830624](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276830624) [](ancestors = 276830624)

Reply:
updated to use sizeof(int32)

---
In reply to: [276865464](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276865464) [](ancestors = 276865464,276830624)